### PR TITLE
add convert subcommand

### PR DESCRIPTION
**Pull request by strongjz**
Open sourcing our internal mconvert tools for managing all our packages in [wolfi-dev/os](https://github.com/wolfi-dev/os) Started by @rawlingsj :partying_face: and extend by the rest of the team, python @strongjz gems @found-it

### CODE REVIEW

File: pkg/python/python.go [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+	// transitive dependencies.
+	ToCheck []string
+}
+
+// TODO: Why isn't this pulled in from melange/build?
+type ImageContents struct {
+	Repositories []string `yaml:"repositories,omitempty"`
+	Keyring      []string `yaml:"keyring,omitempty"`
+	Packages     []string `yaml:"packages,omitempty"`
+}
```
Comment (found-it) [inline pkg/python/python.go:79]: This can likely be removed since it's in the same repository now

File: pkg/python/python.go [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+
+	context := PythonContext{
+		PackageName: packageName,
+		Logger:      log.New(log.Writer(), "convert:python: ", log.LstdFlags|log.Lmsgprefix),
+		ToGenerate:  make(map[string]Package),
+	}
+	return context, nil
+}
+
+// Generate is the entrypoint to generate a ruby gem melange file. It handles
```
Comment (found-it) [inline pkg/python/python.go:96]: Looks like the python comments are still referring to ruby

**Review by found-it — CHANGES_REQUESTED** [review]

File: pkg/python/python.go (hunk 1 above)
Comment (strongjz) [reply pkg/python/python.go:79]: its still apko_types.ImageConfiguration if im not mistaken

Comment (found-it) [reply pkg/python/python.go:79]: Ah yes, I think that TODO is wrong. I do remember for some reason I had issues importing ImageContents from apko types but I was likely doing something wrong and settled with this workaround

File: pkg/convert/wolfios/wolfi-os.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
+package wolfios
```
Comment (found-it) [inline pkg/convert/wolfios/wolfi-os.go:1]: Can we break this package out from under convert/apkbuild like we did with manifest and http?

File: pkg/convert/convert_test.go [hunk 4]
```diff
@@ -0,0 +1,331 @@
+package convert
```
Comment (found-it) [inline pkg/convert/convert_test.go:1]: Can we rename this package to `apkbuild` to make it consistent with python and gem?

**Review by found-it — CHANGES_REQUESTED** [review]

File: pkg/convert/convert_test.go (hunk 4 above)
Comment (strongjz) [reply pkg/convert/convert_test.go:1]: it is an apkbuild file; if anything we should probably change python to pypi ?

File: pkg/convert/wolfios/wolfi-os.go (hunk 3 above)
Comment (strongjz) [reply pkg/convert/wolfios/wolfi-os.go]: done

**Comment by tuananh** [comment]
hoooray !!!!

**Comment by kaniini** [comment]
It would be nice to put all of the convert-related packages under pkg/convert as subpackages.

**Comment by strongjz** [comment]
> 

So I understand, move python, apkbuild, and gem in pkg/convert/X ?

**Comment by kaniini** [comment]
> So I understand, move python, apkbuild, and gem in pkg/convert/X ?

Yes!

**Review by kaniini — APPROVED** [review]

**Review by found-it — APPROVED** [review]
Looks good to me!

**Pull request merged**